string connected_to
string through_door
